pandas/core/groupby/generic.py

```diff
@@ -261,17 +268,46 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             if self.grouper.nkeys > 1:
                 return self._python_agg_general(func, *args, **kwargs)
 
-            try:
-                return self._python_agg_general(func, *args, **kwargs)
-            except (ValueError, KeyError):
-                # TODO: KeyError is raised in _python_agg_general,
-                #  see see test_groupby.test_basic
-                result = self._aggregate_named(func, *args, **kwargs)
+            if isinstance(
+                self._selected_obj.index, (DatetimeIndex, TimedeltaIndex, PeriodIndex)
```

Comment:
Are these changes meant to stay here or part of the other PR you have to clean up groupby?

Reply:
Well the other one is pretty much just to make this work, so it could be closed and just do this

(also this needs to be rebased on a newer commit in that PR that cleans up 286-306 below)